Objects/frameobject.c

```diff
@@ -1804,11 +1804,22 @@ PyDoc_STRVAR(clear__doc__,
 static PyObject *
 frame_sizeof(PyFrameObject *f, PyObject *Py_UNUSED(ignored))
 {
-    Py_ssize_t res;
-    res = offsetof(PyFrameObject, _f_frame_data) + offsetof(_PyInterpreterFrame, localsplus);
+    Py_ssize_t res = offsetof(PyFrameObject, _f_frame_data)
+                     + offsetof(_PyInterpreterFrame, localsplus);
     PyCodeObject *code = _PyFrame_GetCode(f->f_frame);
-    res += _PyFrame_NumSlotsForCodeObject(code) * sizeof(PyObject *);
-    return PyLong_FromSsize_t(res);
+    int nslots = _PyFrame_NumSlotsForCodeObject(code);
+    assert(nslots >= 0);
+    if ((size_t)nslots >= (PY_SSIZE_T_MAX - res) / sizeof(PyObject *)) {
```

Comment:
I think this is UB. Signed integer overflow is undefined.

Reply:
What would be UB? nslots is clipped to an int either way but we need to ensure that `nslots * sizeof(PyObject *)` does not overflow an *int* and not a `Py_ssize_t` (that's the issue IMO). 

The reason is due to the `assert(size < INT_MAX/sizeof(PyObject *));` in `_PyThreadState_PushFrame`.

Summary:

- `co_stacksize` is an int and thus is bounded by INT_MAX
- `co_framesize` is a function of `co_stacksize` but still < INT_MAX
-  `nslots` < `co_framesize` but we don't `sizeof(PyObject *) * nslots + res` to exceed a ssize_t

IIRC, POSIX doesn't specify that `sizeof(int) <= sizeof(ssize_t)` (nor does it guarantee `>=`). Since I don't really want to change the assertion on the size because otherwise this could mess up other parts of the code, we should make sure that the sizeof won't overflow.

Well.. technically, my fix is just about `con->stacksize >= (INT_MAX - nlocalsplus - FRAME_SPECIALS_SIZE) / sizeof(PyObject *)` but it didn't work for generator's frames actually.

Reply:
If I'm understanding this right, we probably need to do something a little more complicated. I don't think it's even *possible* to check if something exceeds the `ssize_t` limit with > or <, because again signed integer overflow is undefined behavior. We probably need to cast things to uint64 and figure it out that way.

Reply:
Actually, I changed that one to an assertion.

Reply:
We are not checking the `ssize_t` limit actually. The assertion I want to avoid is `size < INT_MAX/sizeof(PyObject *)` not `size < PY_SSIZE_T_MAX/sizeof(PyObject *)`. 

You can *in advance* determine whether `X * Y` would overflow if you know both `X` and `Y` and the upper limit you want to impose on the output.